File: CentralIT/src/app/models/request/requestGet.ts
import { DepartmentModel } from "../masters/department";
import { PositionModel } from "../masters/position";
import { PriorityModel } from "../masters/priority";
import { RequestTypeModel } from "../masters/requesttype";
import { ServiceModel } from "../masters/service";
import { StatusModel } from "../masters/status";
import { UserModel } from "../security/user";

export class RequestGetModel{
    requestId: number = -1;
    nameUserRequested: string = "";
    userAttendedId: number = -1;
    nameUserAttended: string = "";
    userRequested: UserModel = new UserModel();
    userAttended: UserModel = new UserModel();
    nameUserRequestedInitial: string = "";
    phoneUserRequestedInitial: string = "";
    requestStatus: StatusModel = new StatusModel();
    requestType: RequestTypeModel = new RequestTypeModel();
    priority: PriorityModel = new PriorityModel();
    service: ServiceModel = new ServiceModel();
    department: DepartmentModel = new DepartmentModel();
    position: PositionModel = new PositionModel();
    reason: string = "";
    observation: string = "";
    status: boolean = false;
    requestDate: Date = new Date();
    attendedDate: Date = new Date();
    finishDate: Date = new Date();
}

File: CentralIT API/CentralIT.Domain/Request/RequestGet.cs
using System;
using System.Collections.Generic;
using System.Text;

using CentralIT.Domain.Common;
using CentralIT.Domain.Department;
using CentralIT.Domain.Position;
using CentralIT.Domain.Service;
using CentralIT.Domain.Users;

namespace CentralIT.Domain.Request {
    public class RequestGet {
        public int RequestId { get; set; }

        public string NameUserRequested { get; set; }

        public int UserAttendedId { get; set; }

        public string NameUserAttended { get; set; }

        public User UserRequested { get; set; }
        
        public User UserAttended { get; set; }

        public string NameUserRequestedInitial { get; set; }

        public string PhoneUserRequestedInitial { get; set; }

        public Status RequestStatus { get; set; }

        public RequestType RequestType { get; set; }

        public Priority Priority { get; set; }

        public Services Service { get; set; }

        public Departments Department { get; set; }

        public Positions Position { get; set; }

        public string Reason { get; set; }

        public string Observation { get; set; }

        public bool Status { get; set; }

        public DateTime RequestDate { get; set; }

        public DateTime AttendedDate { get; set; }

        public DateTime FinishDate { get; set; }

    }
}
